src/plugins/core/tables.ts

```diff
@@ -130,24 +125,25 @@ export class TablePlugin extends CorePlugin<TableState> implements TableState {
         this.dispatch("REMOVE_MERGE", { sheetId: cmd.sheetId, target: mergesInTarget });
 
         const id = this.uuidGenerator.uuidv4();
-        const newTable = this.createTable(id, union, cmd.config || DEFAULT_TABLE_CONFIG);
+        const config = cmd.config || DEFAULT_TABLE_CONFIG;
+        const newTable = cmd.isDynamic
+          ? this.createDynamicTable(id, union, config)
+          : this.createStaticTable(id, union, config);
         this.history.update("tables", cmd.sheetId, newTable.id, newTable);
         break;
       }
       case "REMOVE_TABLE": {
-        const tables: Record<UID, Table> = {};
-        for (const table of this.getTables(cmd.sheetId)) {
-          if (cmd.target.every((zone) => !intersection(zone, table.range.zone))) {
+        const tables: Record<UID, InternalTable> = {};
+        for (const table of this.getInternalTables(cmd.sheetId)) {
```

Comment:
It was already present before but this command can technically delete several tables at a time. should the command have been targeting the exact table zone definition all along?

Reply:
Probably could have been a good way to do it, but at the time it would have required a bit more logic in the menu items for no real reason.

Now I'm stuck with it, I don't see how I would migrate the revisions correctly in python.